src/test/java/org/jabref/model/texparser/CitationTest.java

```diff
@@ -0,0 +1,118 @@
+
+public class CitationTest {
+
+    Path path;
+    Citation citation;
+
+    @BeforeEach
+    public void init() {
+        path = Path.of("test");
+        citation = new Citation(path, 10, 1, 4, "lineText");
+    }
+
+    private static Stream<Arguments> colStartColEndNotInBounds() {
+        return Stream.of(
+                arguments(-1, 2),
+                arguments(1, 9)
+        );
+    }
+
+    private static Stream<Arguments> colStartColEndInBounds() {
+        return Stream.of(
+                arguments(0, 2),
+                arguments(1, 8)
+        );
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {-1, 0})
+    public void constructorLineSmallerEqualZeroTest(int line) {
+        Exception e = assertThrows(IllegalArgumentException.class, () -> new Citation(path, line, 1, 5, "lineText"));
+        assertEquals("Line has to be greater than 0.", e.getMessage());
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {1, 2})
+    public void constructorLineLargerZeroTest(int line) {
+        Citation citation = new Citation(path, line, 1, 5, "lineText");
+    }
+
+    @ParameterizedTest
+    @MethodSource("colStartColEndNotInBounds")
+    public void constructorColStartColEndNotInBoundsTest(int colStart, int colEnd) {
+        Exception e = assertThrows(IllegalArgumentException.class, () -> new Citation(path, 10, colStart, colEnd, "lineText"));
+        assertEquals("Citation has to be between 0 and line length.", e.getMessage());
+    }
+
+    @ParameterizedTest
+    @MethodSource("colStartColEndInBounds")
+    public void constructorColStartColEndInBoundsTest(int colStart, int colEnd) {
+        Citation citation = new Citation(path, 10, colStart, colEnd, "lineText");
+    }
+
+    @Test
+    public void getPathTest() {
+        assertEquals(path, citation.getPath());
+    }
+
+    @Test
+    public void getLineTest() {
+        assertEquals(10, citation.getLine());
+    }
+
+    @Test
+    public void getColStartTest() {
+        assertEquals(1, citation.getColStart());
+    }
+
+    @Test
+    public void getColEndTest() {
+        assertEquals(4, citation.getColEnd());
+    }
+
+    @Test
+    public void getLineTextTest() {
+        assertEquals("lineText", citation.getLineText());
+    }
+
+    @Test
+    public void getContextTest() {
+        assertEquals("lineText", citation.getContext());
+    }
+
+    @Test
+    public void toStringTest() {
+        assertEquals("Citation{path=test, line=10, colStart=1, colEnd=4, lineText='lineText'}", citation.toString());
+    }
+
+    @Test
+    public void equalsTest() {
+        Citation citation1 = new Citation(path, 10, 1, 4, "lineText");
+        Citation citation2 = null;
+        assertTrue(citation.equals(citation1));
+        assertFalse(citation.equals(citation2));
+    }
+
+    @Test
+    public void hashCodeTest() {
+        assertEquals(Objects.hash(path, 10, 1, 4, "lineText"), citation.hashCode());
```

Comment:
Tests for Hashcode and toString do not really add any value and especially toString is more of an internal representation.

Reply:
Oh ok, seemed like a rather easy way to increase the code coverage. Thanks for letting me know, I'll try to avoid them in them in the future. (I removed them with commit [21050ff](https://github.com/JabRef/jabref/pull/7542/commits/21050ff5a9a9451fbde213c3cb00dd9a4fcf5053))